pynest/nest/tests/test_sp/test_growth_curves.py

```diff
@@ -339,22 +339,28 @@ def simulate(self):
                     self.se_nest[n_i], self.se_python[sei_i], decimal=5)
 
     def plot(self):
-        pylab.ion()
-        for i, sei in enumerate(self.se_integrator):
-            pylab.figure()
-            pylab.subplot(1, 2, 1)
-            pylab.title('Ca')
-            pylab.plot(self.sim_steps, self.ca_nest[0, :])
-            pylab.plot(self.sim_steps, self.ca_python[i])
-            pylab.legend(('nest', sei.__class__.__name__))
-            pylab.subplot(1, 2, 2)
-            pylab.title('Synaptic Element')
-            pylab.plot(self.sim_steps, self.se_nest[0, :])
-            pylab.plot(self.sim_steps, self.se_python[i])
-            pylab.legend(('nest', sei.__class__.__name__))
-            pylab.savefig('sp' + sei.__class__.__name__ + '.png')
-        raster_plot.from_device(self.spike_detector)
-        pylab.savefig('sp_raster_plot.png')
+        try:
+            import pylab
+            from nest import raster_plot
+        except ImportError:
+            pass
+        else:
+            pylab.ion()
+            for i, sei in enumerate(self.se_integrator):
+                pylab.figure()
+                pylab.subplot(1, 2, 1)
+                pylab.title('Ca')
+                pylab.plot(self.sim_steps, self.ca_nest[0, :])
+                pylab.plot(self.sim_steps, self.ca_python[i])
+                pylab.legend(('nest', sei.__class__.__name__))
+                pylab.subplot(1, 2, 2)
+                pylab.title('Synaptic Element')
+                pylab.plot(self.sim_steps, self.se_nest[0, :])
+                pylab.plot(self.sim_steps, self.se_python[i])
+                pylab.legend(('nest', sei.__class__.__name__))
+                pylab.savefig('sp' + sei.__class__.__name__ + '.png')
+            raster_plot.from_device(self.spike_detector)
+            pylab.savefig('sp_raster_plot.png')
```

Comment:
Why should we plot in a test at all?

Reply:
Look at [this line](https://github.com/tammoippen/nest-simulator/blob/6c71b55ff9b6e3bb154f3ee5683f20039533d356/pynest/nest/tests/test_sp/test_growth_curves.py#L489). Apparently, the author of the test wanted to debug the tests...

Reply:
@tammoippen I suppose this is a matter of philosophy, but it makes some sense. To make the code more legible, I would suggest 
- to move the definition of the `plot()` method to right before
- rename it `_plot()` to mark it as internal
- add a docstring like `"Can be called by tearDown() to cross-check tests.`.

Reply:
@heplesser 
> * to move the definition of the plot() method to right before

Which version are you referring to? The complete `_plot` code in the `try:` block?